--- CRT.c.orig	2020-12-31 09:55:03.341869750 +0800
+++ CRT.c	2020-12-31 09:57:39.881184213 +0800
@@ -125,7 +125,7 @@
       [LED_COLOR] = ColorPair(Green, Black),
       [TASKS_RUNNING] = A_BOLD | ColorPair(Green, Black),
       [PROCESS] = A_NORMAL,
-      [PROCESS_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [PROCESS_SHADOW] = ColorPairGrayBlack,
       [PROCESS_TAG] = A_BOLD | ColorPair(Yellow, Black),
       [PROCESS_MEGABYTES] = ColorPair(Cyan, Black),
       [PROCESS_GIGABYTES] = ColorPair(Green, Black),
@@ -142,7 +142,7 @@
       [PROCESS_COMM] = ColorPair(Magenta, Black),
       [PROCESS_THREAD_COMM] = ColorPair(Blue, Black),
       [BAR_BORDER] = A_BOLD,
-      [BAR_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [BAR_SHADOW] = ColorPairGrayBlack,
       [SWAP] = ColorPair(Red, Black),
       [GRAPH_1] = A_BOLD | ColorPair(Cyan, Black),
       [GRAPH_2] = ColorPair(Cyan, Black),
@@ -166,7 +166,7 @@
       [CPU_NICE_TEXT] = A_BOLD | ColorPair(Blue, Black),
       [CPU_NORMAL] = ColorPair(Green, Black),
       [CPU_SYSTEM] = ColorPair(Red, Black),
-      [CPU_IOWAIT] = A_BOLD | ColorPairGrayBlack,
+      [CPU_IOWAIT] = ColorPairGrayBlack,
       [CPU_IRQ] = ColorPair(Yellow, Black),
       [CPU_SOFTIRQ] = ColorPair(Magenta, Black),
       [CPU_STEAL] = ColorPair(Cyan, Black),
@@ -383,7 +383,7 @@
       [LED_COLOR] = ColorPair(Green, Black),
       [TASKS_RUNNING] = ColorPair(Green, Black),
       [PROCESS] = ColorPair(Black, Black),
-      [PROCESS_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [PROCESS_SHADOW] = ColorPairGrayBlack,
       [PROCESS_TAG] = ColorPair(White, Blue),
       [PROCESS_MEGABYTES] = ColorPair(Blue, Black),
       [PROCESS_GIGABYTES] = ColorPair(Green, Black),
@@ -550,7 +550,7 @@
       [LED_COLOR] = ColorPair(Green, Black),
       [TASKS_RUNNING] = A_BOLD | ColorPair(Green, Black),
       [PROCESS] = ColorPair(Cyan, Black),
-      [PROCESS_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [PROCESS_SHADOW] = ColorPairGrayBlack,
       [PROCESS_TAG] = A_BOLD | ColorPair(Yellow, Black),
       [PROCESS_MEGABYTES] = A_BOLD | ColorPair(Green, Black),
       [PROCESS_GIGABYTES] = A_BOLD | ColorPair(Yellow, Black),
@@ -668,7 +668,7 @@
 
    for (int i = 0; i < LAST_COLORELEMENT; i++) {
       unsigned int color = CRT_colorSchemes[COLORSCHEME_DEFAULT][i];
-      CRT_colorSchemes[COLORSCHEME_BROKENGRAY][i] = color == (A_BOLD | ColorPairGrayBlack) ? ColorPair(White, Black) : color;
+      CRT_colorSchemes[COLORSCHEME_BROKENGRAY][i] = color == (ColorPairGrayBlack) ? ColorPair(White, Black) : color;
    }
 
    halfdelay(*CRT_delay);
@@ -801,7 +801,7 @@
       }
    }
 
-   short int grayBlackFg = COLORS > 8 ? 8 : 0;
+   short int grayBlackFg = COLORS > 8 ? 15 : 7;
    short int grayBlackBg = (colorScheme != COLORSCHEME_BLACKNIGHT) ? -1 : 0;
    init_pair(ColorIndexGrayBlack, grayBlackFg, grayBlackBg);
 
