--- CRT.c.orig	2018-04-10 21:23:24.000000000 +0800
+++ CRT.c	2018-08-09 21:43:38.619653456 +0800
@@ -193,7 +193,7 @@
       [LED_COLOR] = ColorPair(Green,Black),
       [TASKS_RUNNING] = A_BOLD | ColorPair(Green,Black),
       [PROCESS] = A_NORMAL,
-      [PROCESS_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [PROCESS_SHADOW] = ColorPairGrayBlack,
       [PROCESS_TAG] = A_BOLD | ColorPair(Yellow,Black),
       [PROCESS_MEGABYTES] = ColorPair(Cyan,Black),
       [PROCESS_BASENAME] = A_BOLD | ColorPair(Cyan,Black),
@@ -205,7 +205,7 @@
       [PROCESS_THREAD] = ColorPair(Green,Black),
       [PROCESS_THREAD_BASENAME] = A_BOLD | ColorPair(Green,Black),
       [BAR_BORDER] = A_BOLD,
-      [BAR_SHADOW] = A_BOLD | ColorPairGrayBlack,
+      [BAR_SHADOW] = ColorPairGrayBlack,
       [SWAP] = ColorPair(Red,Black),
       [GRAPH_1] = A_BOLD | ColorPair(Cyan,Black),
       [GRAPH_2] = ColorPair(Cyan,Black),
@@ -227,7 +227,7 @@
       [CPU_NICE_TEXT] = A_BOLD | ColorPair(Blue,Black),
       [CPU_NORMAL] = ColorPair(Green,Black),
       [CPU_KERNEL] = ColorPair(Red,Black),
-      [CPU_IOWAIT] = A_BOLD | ColorPair(Black, Black),
+      [CPU_IOWAIT] = ColorPairGrayBlack,
       [CPU_IRQ] = ColorPair(Yellow,Black),
       [CPU_SOFTIRQ] = ColorPair(Magenta,Black),
       [CPU_STEAL] = ColorPair(Cyan,Black),
@@ -726,11 +726,18 @@
       }
    }
 
-   int grayBlackFg = COLORS > 8 ? 8 : 0;
+   int grayBlackFg = COLORS > 8 ? 15 : 7;
    int grayBlackBg = (colorScheme != COLORSCHEME_BLACKNIGHT)
                      ? -1
                      : 0;
    init_pair(ColorIndexGrayBlack, grayBlackFg, grayBlackBg);
 
    CRT_colors = CRT_colorSchemes[colorScheme];
+
+   if (COLORS <= 8) {
+       for (int i = 0; i < LAST_COLORELEMENT; ++i) {
+           if (CRT_colors[i] == ColorPairGrayBlack)
+               CRT_colors[i] = A_BOLD | CRT_colors[i];
+       }
+   }
 }
