dif --git a/latexrun b/latexrun
--- a/latexrun	2017-05-27 15:25:53.133233603 -0300
+++ b/latexrun	2017-05-27 15:26:39.572774132 -0300
@@ -1290,10 +1290,15 @@
                 self.__message('warning', None,
                                "extra `)' in log; file names may be wrong ")
         elif ch == '{':
-            # TeX uses this for various things we want to ignore, like
-            # file names and print_mark.  Consume up to the '}'
-            epos = self.__data.find('}', self.__pos)
-            if epos != -1:
+            epos = self.__pos
+            count = 1
+            while epos < len(self.__data) and count > 0:
+                if self.__data[epos] == '}':
+                    count -= 1
+                elif self.__data[epos] == '{':
+                    count += 1
+                epos += 1
+            if epos != -1 and count == 0:
                 self.__pos = epos + 1
             else:
                 self.__message('warning', None,
