--- a/alot/widgets/search.py	2018-03-27 16:14:06.887154993 +0800
+++ a/alot/widgets/search.py	2018-03-27 16:13:57.418136838 +0800
@@ -77,7 +77,7 @@
             else:
                 authors = '(None)'
             authorsstring = pad(authors, shorten_author_string)
-            authors_w = AttrFlipWidget(urwid.Text(authorsstring),
+            authors_w = AttrFlipWidget(urwid.Text(authorsstring, wrap='clip'),
                                        struct['authors'])
             width = len(authorsstring)
             part = authors_w
