--- x.c	2018-01-02 09:31:24.971003021 +0800
+++ x.c	2018-01-02 10:02:03.844102008 +0800
@@ -1199,9 +1199,9 @@
 	}
 
 	if ((base.mode & ATTR_BOLD_FAINT) == ATTR_FAINT) {
-		colfg.red = fg->color.red / 2;
-		colfg.green = fg->color.green / 2;
-		colfg.blue = fg->color.blue / 2;
+		colfg.red = (bg->color.red + 2*fg->color.red) / 3;
+		colfg.green = (bg->color.green + 2*fg->color.green) / 3;
+		colfg.blue = (bg->color.blue + 2*fg->color.blue) / 3;
 		colfg.alpha = fg->color.alpha;
 		XftColorAllocValue(xw.dpy, xw.vis, xw.cmap, &colfg, &revfg);
 		fg = &revfg;
